cosmovisor/cmd/cosmovisor/cmd/root.go

```diff
@@ -1,44 +1,28 @@
 package cmd
 
 import (
-	"strings"
+	"os"
 
 	"github.com/rs/zerolog"
-)
-
-// RunCosmovisorCommand executes the desired cosmovisor command.
-func RunCosmovisorCommand(logger *zerolog.Logger, args []string) error {
-	arg0 := ""
-	if len(args) > 0 {
-		arg0 = strings.TrimSpace(args[0])
-	}
-
-	switch {
-	case IsVersionCommand(arg0):
-		return PrintVersion(logger, args[1:])
-
-	case ShouldGiveHelp(arg0):
-		return DoHelp()
+	"github.com/spf13/cobra"
 
-	case IsRunCommand(arg0):
-		return Run(logger, args[1:])
-	}
+	cverrors "github.com/cosmos/cosmos-sdk/cosmovisor/errors"
+)
 
-	warnRun := func() {
-		logger.Warn().Msg("use of cosmovisor without the 'run' command is deprecated. Use: cosmovisor run [args]")
-	}
-	warnRun()
-	defer warnRun()
+var logger *zerolog.Logger
```

Comment:
I think we should avoid using a private global logger. Can we instead create the logger in the command handler?

Reply:
Alright, using logger from context instead